### PR TITLE
 Bootstrap project, modules, directories, and placeholder classes

### PR DESCRIPTION
**Pull request by liujoshua**

### CODE REVIEW

**Comment by liujoshua** [comment]
Classes (including Activities, Fragments) and resources are auto generated, just placeholders.

**Comment by liujoshua** [comment]
`./gradlew dokka` will generate JavaDoc for both Kotlin and Java

**Comment by liujoshua** [comment]
Please review the following
- module organization
- package/directory organization
- dependencies/tools

Note that `:core-mobile` is an Android library, and `:sample-app` is an Android application. `:api` and `:core` are Java/Kotlin libraries. `:api` contains interfaces only. A package with only interfaces makes it cleaner to write plug-ins and swappable components.

File: api/src/main/java/org/researchstack/step/Step.kt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package org.researchstack.step
+
+interface Step {
```
Comment (syoung-smallwisdom) [inline api/src/main/java/org/researchstack/step/Step.kt]: rename `IStep`

File: api/src/main/java/org/researchstack/result/Result.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package org.researchstack.result
+
+/**
+ * A Result.
+ */
+interface Result {
```
Comment (syoung-smallwisdom) [inline api/src/main/java/org/researchstack/result/Result.kt]: rename `IResult`

File: api/src/main/java/org/researchstack/task/Task.kt [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package org.researchstack.task
+
+interface Task {
```
Comment (syoung-smallwisdom) [inline api/src/main/java/org/researchstack/task/Task.kt]: rename `ITask`

**Pull request merged**